src/components/Budget/BudgetView/BudgetViewContainer.js

```diff
@@ -74,6 +82,32 @@ const BudgetViewContainer = ({ history, match, mutator }) => {
     [history, budget],
   );
 
+  const removeBudget = useCallback(
+    () => {
+      mutator.budgetById.DELETE({ id: budgetId })
+        .then(() => {
+          showCallout({ messageId: 'ui-finance.budget.actions.remove.success', type: 'success' });
+          history.replace({
+            pathname: `${FUNDS_ROUTE}/view/${budget.fundId}`,
+            search: location.search,
+          });
+        })
+        .catch(() => {
+          showCallout({ messageId: 'ui-finance.budget.actions.remove.error', type: 'error' });
+        });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [location.search, budgetId],
```

Comment:
does it really depends on `location.search`?

Reply:
nope